### PR TITLE
Don't warn unused params for trivial RHS

### LINKED ISSUE #11856: Don't lint unused params for trivial RHS
This doesn't warn:
```
def f(x: Int) = ???
```
Other expressive RHS:
```
def f(x: Int) = ()
def f(x: Int) = throw new RuntimeException
def f(x: Int) = 42
def f(x: Int): Option[Int] = None
```
Narrowly-occupied types indicate an intention to discard inputs. Probably overrides are invited.

### PR DESCRIPTION
**Pull request by som-snytt**
Fixes scala/bug#11856

### CODE REVIEW

File: test/files/neg/warn-unused-params.scala [hunk 1]
```diff
@@ -75,28 +77,31 @@ trait Unimplementation {
...
 }
 trait Context[A]
 trait Implicits {
-  def f[A](implicit ctx: Context[A]) = 42
-  def g[A: Context] = 42
+  def f[A](implicit ctx: Context[A]) = answer
+  def g[A: Context] = answer
 }
 class Bound[A: Context]
+object Answers {
```
Comment (lrytz) [inline test/files/neg/warn-unused-params.scala:105]: this makes it sound like there's more than one 🤔

**Review by lrytz — APPROVED** [review]
nice 👍

File: test/files/neg/warn-unused-params.scala (hunk 1 above)
Comment (som-snytt) [reply test/files/neg/warn-unused-params.scala:105]: I already suggested that by not ascribing the literal type...

Comment (lrytz) [reply test/files/neg/warn-unused-params.scala:105]: I see... So we need to [keep on searching](https://www.youtube.com/watch?v=dTlGZOsvB5Q)

Comment (som-snytt) [reply test/files/neg/warn-unused-params.scala:105]: @lrytz thanks I didn't know that one

**Comment by SethTisue** [comment]
@som-snytt there is a community build regression that, at first glance anyway, appears to be associated with this PR: https://scala-ci.typesafe.com/job/scala-2.13.x-jdk8-integrate-community-build/4954/artifact/logs/sttp-build.log — can you take a look for 2.13.9? (which isn't imminent, since we're waiting for 3.2.0)

this PR was merged 4 months ago, so I don't know why this didn't crop up until now; perhaps the code in sttp itself is new

**Comment by som-snytt** [comment]
"Smart" NPE would be nice right about now.

I'll change the code comment to `so trivial it might be null`.

**Pull request merged**